### PR TITLE
Add plugin: Structured Tree

### PR DESCRIPTION
**Pull request by Rudtrack**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/Rudtrack/structured-tree

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [x]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by joethei** [comment]
What are the actual changes between your plugin and the original?
If it's clearly abandoned we can transfer the ownershipf over the plugin ID over to you, but for that we need the diff to judge wether the plugin significantly deviates from the original.

[Copyright (c) 2023 Levi Rizki Saputra](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/LICENSE#L3)
Update this.

["author": "Marius Svarverud",](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/manifest.json#L7)
Credit the original author here.

[name: "Lookup Note",](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/main.ts#L26), [this.addRibbonIcon(dendronActivityBarName, "Open Structured Tree", () => {](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/main.ts#L36), [.setTitle("Reveal in Structured Tree")](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/main.ts#L160), [.setName("Deletion Method")](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/settings.ts#L47), [.addOption('moveToTrash', 'Move to Trash')](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/settings.ts#L52-L53), [.setName("Auto Generate Front Matter")](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/settings.ts#L61), [.setName("Auto Reveal")](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/settings.ts#L71), [.setName("Custom Resolver")](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/settings.ts#L81), [.setName("Custom Graph Engine")](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/settings.ts#L93), [new Setting(containerEl).setName("Vault List").setHeading();](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/settings.ts#L102), [btn.setButtonText("Add Vault").onClick(() => {](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/settings.ts#L116), [return "Structured Tree";](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/view.ts#L23), [this.contentEl.createEl("h1", { text: "Invalid Root" });](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/modal/invalid-root.ts#L10), [new Setting(this.contentEl).setHeading().setName("Add Vault");](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/modal/add-vault.ts#L73-L74), [new Setting(this.contentEl).setName("Vault Name").addText((text) => {](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/modal/add-vault.ts#L86), [.setButtonText("Add Text")](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/modal/add-vault.ts#L92), [new Notice("Please specify Vault Path and Vault Name");](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/modal/add-vault.ts#L96), [item.setTitle("Create Current Note").setIcon("create-new").onClick(createCurrentNote);](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/components/NoteComponent.svelte#L80), [item.setTitle("Create New Note").setIcon("plus").onClick(openLookup);](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/components/NoteComponent.svelte#L85), [item.setTitle("Delete Note").setIcon("trash").onClick(deleteCurrentNote);](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/components/NoteComponent.svelte#L90)


[.setName("Auto Generate Front Matter")](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/settings.ts#L61)
It's called frontmatter

[export const dendronActivityBarIcon](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/icons.ts#L2)
Please don't use the dendron icon, the dendron team has the copyright over it.

[generateFronmatter](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/engine/vault.ts#L57)
Typo

[console.log("Retry replace graph node text");](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/src/custom-graph/index.ts#L33)
Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.

**Comment by Rudtrack** [comment]
Thanks for your quick response, I will fix the issues you mentioned! As for the diff between the original plugin and mine, currently the only major difference is that I've added the setting entry and the logic for choosing how a file is deleted when it's done through the plugin. In the original plugin this was always "Delete Permenantly" with no confirmation required. This can and has lead to accidental deletion of notes with not easy way of recoverying them.

The original plugin has not seen any activity for over 10 months. I created a pull request in mid-May with my changes that I described above, but I have yet to get a response. I've tried to get into contact with the author, but they have very little contact information publicly available and they are not responding to efforts through GitHub.

This is a plugin I use daily and is a key aspect of my workflow, so I didn't want to see it get abandonded and I see so much potential for further development with this plugin

**Comment by joethei** [comment]
for the deletion you should use `app.fileManager.trashFile`, it will delete the file according to the _Files and links -> Deleted files_ setting.
Was added in 1.6.5

**Comment by Rudtrack** [comment]
I've implemented the changes you suggested into my repo and created a new release (1.0.4). Do I need to sync the `patch-1` branch, or does something happen automatically?

I got one small question regarding this though:

> ["author": "Marius Svarverud",](https://github.com/Rudtrack/structured-tree/blob/b0df57e54038193a3824d68a66cd17e7235bbcf5/manifest.json#L7)
> Credit the original author here.

Should I keep the original authors URL and funding URL? Currently those links are for me

**Comment by joethei** [comment]
authorUrl and fundingUrl don't need to be kept.

**Comment by Rudtrack** [comment]
Great, thank you! As I said, I've made a new release from my repo. Let me know if I need to do anything else!

**Comment by Rudtrack** [comment]
Hi, any update on this? Like I said, I've made the changes requested and created a new release (1.0.4) from my repo

**Comment by liamcain** [comment]
Hi, sorry for the delay. I'm taking over plugin reviews while joethei is out on sick leave.

- [Lookup Note](https://github.com/Rudtrack/structured-tree/blob/07112f43563f807f56c18defbc7467903994ca18/src/main.ts#L26) [The text here should use 'Sentence case'](<https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI>)
- Your readme has a few references to the original plugin name which might be confusing for users. The command "Open Dendron Tree" doesn't exist, for example.

File: community-plugins.json [hunk 1]
```diff
@@ -12955,5 +12955,11 @@
...
         "description": "Supports rendering of GuitarPro files such as `gtp/gp/gp5/gpx`.",
         "repo": "Barba828/obsidian-plugin-gtp-preview"
-    }
+    },
+    {
+        "id": "structured-tree",
+        "name": "Structured Tree",
+        "author": "Marius Svarverud",
+        "description": "A file explorer for navigating hierarchical notes separated by '.'",
+        "repo": "Rudtrack/structured-tree"
```
Comment (liamcain) [inline community-plugins.json:12964]:
```suggestion
        "repo": "Rudtrack/structured-tree"
}
```

**Pull request merged**